superset/viz.py

```diff
@@ -112,6 +112,16 @@ def get_df(self, query_obj=None):
                 self.error_message = "No data."
             return pd.DataFrame()
         else:
+            if hasattr(self.datasource.database, 'db_engine_spec') and \
```

Comment:
The `db_engine_spec` module is there to keep all of the database-engine-specific code in one place, so that the code base is not littered with `if engine == 'some_engine'` conditions all over the place. Having it all in one place makes it easier to add new engines as well and use inheritance schemes where it makes sense.

In this case you're trying to alter the column name to upper case, and I'm not sure why column that contain `__` are excluded, but somehow this should happen ideally upfront if possible (as we load up the metadata in the columns table). Otherwise you probably want to add a `format_column_name` method to `superset.db_engine_spec.BaseEngineSpec` that does nothing (aka `lambda x: x`), and override that method for `OracleEngineSpec`